pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -194,71 +201,87 @@ func (m *ManifestWorkController) sync(ctx context.Context, controllerContext fac
 }
 
 func (m *ManifestWorkController) applyManifests(
-	manifests []workapiv1.Manifest, recorder events.Recorder, owner metav1.OwnerReference, existingResults []resourceapply.ApplyResult) []resourceapply.ApplyResult {
+	manifests []workapiv1.Manifest, recorder events.Recorder, deleteOption *workapiv1.DeleteOption, owner metav1.OwnerReference, existingResults []applyResult) []applyResult {
 	for index, manifest := range manifests {
 		switch {
 		case existingResults[index].Result == nil:
 			// Apply if there is not result.
-			existingResults[index] = m.applyOneManifest(manifest, recorder, owner)
+			existingResults[index] = m.applyOneManifest(index, manifest, deleteOption, recorder, owner)
 		case errors.IsConflict(existingResults[index].Error):
 			// Apply if there is a resource confilct error.
-			existingResults[index] = m.applyOneManifest(manifest, recorder, owner)
+			existingResults[index] = m.applyOneManifest(index, manifest, deleteOption, recorder, owner)
 		}
 	}
 
 	return existingResults
 }
 
 func (m *ManifestWorkController) applyOneManifest(
-	manifest workapiv1.Manifest, recorder events.Recorder, owner metav1.OwnerReference) resourceapply.ApplyResult {
+	index int, manifest workapiv1.Manifest, deleteOption *workapiv1.DeleteOption, recorder events.Recorder, owner metav1.OwnerReference) applyResult {
 	clientHolder := resourceapply.NewClientHolder().
 		WithAPIExtensionsClient(m.spokeAPIExtensionClient).
 		WithKubernetes(m.spokeKubeclient).
 		WithDynamicClient(m.spokeDynamicClient)
 
+	result := applyResult{}
+
+	resMeta, gvr, err := buildManifestResourceMeta(index, manifest, m.restMapper)
+	result.resourceMeta = resMeta
+	if err != nil {
+		result.Error = err
+		return result
+	}
+
 	results := resourceapply.ApplyDirectly(clientHolder, recorder, func(name string) ([]byte, error) {
```

Comment:
the ApplyDirectly doesn't directly manage ownerReferences?  Should it merge by default, similar to the handling for labels and annotations?

Reply:
ApplyDirectly does not merge ownerReferences today, https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourcemerge/object_merger.go#L12.

We discussed this before that we might add a customized hooks for all apply func, but I haven't done that yet.